### PR TITLE
Fixes tvos sigh profile name

### LINKED ISSUE #8122: Match can not fetch tvos provisioning profiles from repo
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description

`match` can not fetch tvos provisioning profiles from repo:

```
No matching provisioning profiles found for 'AppStore_com.awesome.app_tvos'
```

I believe the problem happens in `sigh` when downloading a tvos provisioning profile. `sigh` is [not append the platform](https://github.com/fastlane/fastlane/blob/4e9fe26be80b9476d5a29e46970c898c91fd2592/sigh/lib/sigh/runner.rb#L219), `_tvos`, to the file name which `match`[ is expecting](https://github.com/bassrock/fastlane/blob/651039ebae7e3adc44714544179539a5ea67fdd4/match/lib/match/runner.rb#L103).

```bash
Creating new provisioning profile for 'com.awesome.app' with name 'match AppStore \
com.awesome.app tvos for tvos platform'
Downloading provisioning profile...
Successfully downloaded provisioning profile...
Installing provisioning profile...
/var/folders/_y/5xvmvfs114g74mfhlkn84fm00000gn/T/d20170203-91989- \
11i20l0/profiles/appstore/AppStore_com.awesome.app.mobileprovision
```

I believe the file should be named AppStore_com.awesome.app_**tvos**.mobileprovision to please `match`.

### Environment

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                 | Value                                       |
| ------------------- | ------------------------------------------- |
| OS                  | 10.12.2                                     |
| Ruby                | 2.3.1                                       |
| Bundler?            | false                                       |
| Git                 | git version 2.10.1 (Apple Git-78)           |
| Installation Source | /usr/local/bin/fastlane                     |
| Host                | Mac OS X 10.12.2 (16C67)                    |
| Ruby Lib Dir        | /usr/local/Cellar/ruby/2.3.1_2/lib          |
| OpenSSL Version     | OpenSSL 1.0.2j  26 Sep 2016                 |
| Is contained        | false                                       |
| Is homebrew         | false                                       |
| Xcode Path          | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version       | 8.2.1                                       |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.14.2  | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version |
| ------------------------- | ------- |
| did_you_mean              | 1.0.0   |
| slack-notifier            | 1.5.1   |
| rouge                     | 1.11.1  |
| xcpretty                  | 0.2.4   |
| terminal-notifier         | 1.7.1   |
| multipart-post            | 2.0.0   |
| word_wrap                 | 1.0.0   |
| babosa                    | 1.0.2   |
| colored                   | 1.2     |
| excon                     | 0.54.0  |
| http-cookie               | 1.0.3   |
| faraday-cookie_jar        | 0.0.6   |
| highline                  | 1.7.8   |
| mini_magick               | 4.5.1   |
| multi_json                | 1.12.1  |
| security                  | 0.1.3   |
| xcpretty-travis-formatter | 0.0.4   |
| bundler                   | 1.13.5  |
| json                      | 2.0.2   |
| commander                 | 4.4.3   |
| io-console                | 0.4.5   |
| rubyzip                   | 1.2.0   |
| plist                     | 3.2.0   |
| claide                    | 1.0.1   |
| nanaimo                   | 0.2.3   |
| xcodeproj                 | 1.4.2   |
| terminal-table            | 1.7.3   |
| unicode-display_width     | 1.1.3   |
</details>

*generated on:* **2017-02-03**
</details>

### PR DESCRIPTION
**Pull request by cgoldsby**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Description
When downloading a profile`sigh` will now append `_tvos` to the file name.

### Motivation and Context
Fixes #8122. This fixes a problem where `match` can not fetch a tvos profile from a repo because the profile name is not found. Feel free to tweak or discard. Or, let me know if you would like me to make any changes. ❤️

### CODE REVIEW

File: sigh/lib/sigh/runner.rb [hunk 1]
```diff
@@ -127,7 +127,7 @@ def create_profile!
         end
       end
 
-      UI.important "Creating new provisioning profile for '#{Sigh.config[:app_identifier]}' with name '#{name} for #{Sigh.config[:platform]} platform'"
+      UI.important "Creating new provisioning profile for '#{Sigh.config[:app_identifier]}' with name '#{name}' for '#{Sigh.config[:platform]}' platform"
```
Comment (hjanuschka) [inline sigh/lib/sigh/runner.rb]: ❤️

File: sigh/lib/sigh/runner.rb [hunk 2]
```diff
@@ -216,7 +216,12 @@ def certificate_to_use
     # Downloads and stores the provisioning profile
     def download_profile(profile)
       UI.important "Downloading provisioning profile..."
-      profile_name ||= "#{profile_type.pretty_type}_#{Sigh.config[:app_identifier]}.mobileprovision" # default name
+      profile_name ||= "#{profile_type.pretty_type}_#{Sigh.config[:app_identifier]}"
```
Comment (hjanuschka) [inline sigh/lib/sigh/runner.rb]: can we append the .mobileprovision directly in the `|| =`  above?
as right now it seems to be appended anyway 

the prev. implementation also was not 100% right, i think

**Review by hjanuschka — CHANGES_REQUESTED** [review]
awesome, good explanation in the issue, nice catch,  thx for your contribution, i left a  note. once that done i think we are good to merge.

File: sigh/lib/sigh/runner.rb (hunk 2 above)
Comment (cgoldsby) [inline sigh/lib/sigh/runner.rb]: @hjanuschka, thank you for the review. We need '_tvos' just before .mobileprovision. So, I am not sure if we can append it to [line 219](https://github.com/fastlane/fastlane/pull/8123/files#diff-e9e7336e24f66a051f7c7e07128efd8fR219). 

Would something like this work for you? Or, did you have something else in mind.

```diff ruby
profile_name ||= "#{profile_type.pretty_type}_#{Sigh.config[:app_identifier]}"
if Sigh.config[:platform].to_s == 'tvos'
    profile_name += "_tvos"
end
- profile_name += '.mobileprovision' unless profile_name.include? 'mobileprovision
+ profile_name += '.mobileprovision'
```

Comment (hjanuschka) [inline sigh/lib/sigh/runner.rb]: ok good point, yep thats fine, i was just curious about the infact usless unless 👍

Comment (cgoldsby) [inline sigh/lib/sigh/runner.rb]: 😎 

✓ Done.

File: sigh/lib/sigh/runner.rb [hunk 3]
```diff
@@ -216,8 +216,13 @@ def certificate_to_use
...
       UI.important "Downloading provisioning profile..."
-      profile_name ||= "#{profile_type.pretty_type}_#{Sigh.config[:app_identifier]}.mobileprovision" # default name
-      profile_name += '.mobileprovision' unless profile_name.include? 'mobileprovision'
+      profile_name ||= "#{profile_type.pretty_type}_#{Sigh.config[:app_identifier]}"
+
+      if Sigh.config[:platform].to_s == 'tvos'
+        profile_name += "_tvos"
+      end
+
+      profile_name += '.mobileprovision'
```
Comment (KrauseFx) [inline sigh/lib/sigh/runner.rb]: Shouldn't this also need ` unless profile_name.include? 'mobileprovision'` like we've had before?

Comment (hjanuschka) [inline sigh/lib/sigh/runner.rb]: ohhh .... i request to remove this, as i thought its not needed as it is always true - @KrauseFx  you mean there is a case where profile_name could already contain `.mobileprovision` - i didn't see any possibilities for that. (in sigh/runner)

Comment (hjanuschka) [inline sigh/lib/sigh/runner.rb]: as `profile_name` is created and only used in the `download_profile` method.

Comment (KrauseFx) [inline sigh/lib/sigh/runner.rb]: Sounds good to me

**Review by hjanuschka — APPROVED** [review]

**Comment by hjanuschka** [comment]
@cgoldsby many thx for your contribution!

**Pull request merged**